dispatcher/backend/src/routes/users/keys.py

```diff
@@ -30,12 +33,20 @@ def get(self, username: str, token: AccessToken.Payload):
             if not token.get_permission("users", "ssh_keys"):
                 raise errors.NotEnoughPrivilege()
 
-        user = Users().find_one({"username": username}, {"ssh_keys": 1})
-        if user is None:
-            raise errors.NotFound()
+        # find user based on username
+        with Session.begin() as session:
+            orm_user = session.execute(
+                sa.select(dbm.User)
+                .where(dbm.User.username == username)
+                .options(so.selectinload(dbm.User.ssh_keys))
+            ).scalar_one_or_none()
 
-        ssh_keys = user.get("ssh_keys", [])
-        return jsonify(ssh_keys)
+            if orm_user is None:
+                raise errors.NotFound()
+
+            api_ssh_keys = list(map(cso.SshKeyRead().dump, orm_user.ssh_keys))
```

Comment:
Do you mean returning directly the `orm_user.ssh_keys` list ? You can't do that due to recursion issues, you have to implement a mapper.
But yes, we could return directly from within the session with `return list(map(cso.SshKeyRead().dump, orm_user.ssh_keys))` and I agree this is way more clear, or is even not relevant anymore once we use a decorator instead of an explicit `with ...` statement. Debug info makes little difference.

Reply:
Yes, that's what I meant. Glad to hear it would work